File: JavaTest/src/IOCDomain/StockQuote.java
package IOCDomain;

import com.google.inject.Inject;

public class StockQuote implements IStockQuote {
    private ILogger _logger;
    private IErrorHandler _handler;
    private IDatabase _database;

    @Inject
    public StockQuote(ILogger logger, IErrorHandler handler, IDatabase database) {
        _logger = logger;
        _handler = handler;
        _database = database;
    }

    @Override
    public ILogger getLogger() {
        return _logger;
    }

    @Override
    public IErrorHandler getErrorHandler() {
        return _handler;
    }

    @Override
    public IDatabase getDatabase() {
        return _database;
    }
}

File: IOCPerfTest/IOCDomain/StockQuote.cs
namespace IOCPerfTest.IOCDomain
{
	public class StockQuote : IStockQuote
	{
	    readonly ILogger _logger;
	    readonly IErrorHandler _handler;
	    readonly IDatabase _database;

		public StockQuote(ILogger logger, IErrorHandler handler, IDatabase database)
		{
			_logger = logger;
			_handler = handler;
			_database = database;
		}

		public ILogger Logger { get { return _logger; } }
		public IErrorHandler ErrorHandler { get { return _handler; } }
		public IDatabase Database { get { return _database; } }
	}
}
